modules/dns-response-policy/main.tf

```diff
@@ -15,7 +15,9 @@
  */
 
 locals {
-  _factory_rules = try(yamldecode(file(var.rules_file)), {})
+  _factory_data  = var.rules_file != null ? file(var.rules_file) : "{}"
```

Comment:
@ludoo @juliocc PTAL.

Reply:
Background: the ternary operator complained for inconsistent type only for some specific yaml files (I have isolated a test case if you're curious). 
To avoid piling up workarounds I'm leveraging a temporary string variable where I store the content of the file (or the representation of an empty yml), and yamldecode() that string after.

Reply:
if this works, then probably `yamldecode(var.rules_file != null ? ... : ...)` should work too? (and I guess, should be default approach where ternary operator is accompanied by `yamldecode`)

Reply:
been through the same hell, would be very curious to see the test case together